presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -116,8 +119,8 @@
     private final Duration maxRetryTime;
     private final int maxRetries;
 
-    private volatile boolean metastoreKnownToSupportTableParamEqualsPredicate;
-    private volatile boolean metastoreKnownToSupportTableParamLikePredicate;
+    private final AtomicInteger verifiedGetTableAlternatives = new AtomicInteger();
+    private final AtomicInteger verifiedTableParamAlternatives = new AtomicInteger();
```

Comment:
can we extract this to some testable entity with a clear contract (API)

Reply:
It wasn't considered as needed before. I'd leave this here for now. I don't want this thin abstraction to blow up